src/sage/modules/with_basis/representation.py

```diff
@@ -475,132 +713,709 @@ def product_by_coercion(self, left, right):
+        category = AlgebrasWithBasis(R).or_subcategory(category)
+        Representation_Exterior.__init__(self, rep, degree=degree, category=category, **options)
+
+    @cached_method
+    def one_basis(self):
+        r"""
+        Return the basis element indexing `1` in ``self`` if it exists.
+
+        EXAMPLES::
+
+            sage: S3 = SymmetricGroup(3)
+            sage: L = S3.regular_representation(side="left")
+            sage: E = L.exterior_power()
+            sage: E.one_basis()
+            0
+            sage: E0 = L.exterior_power(0)
+            sage: E0.one_basis()
+            0
+        """
+        return self._indices([])
+
+    def product_on_basis(self, x, y):
+        r"""
+        Return the product of basis elements indexed by ``x`` and ``y``.
+
+        EXAMPLES::
+
+            sage: S3 = SymmetricGroup(3)
+            sage: L = S3.regular_representation(side="left")
+            sage: E = L.exterior_power()
+            sage: B = list(E.basis())
+            sage: B[:7]
+            [1, (), (1,3,2), (1,2,3), (2,3), (1,3), (1,2)]
+            sage: B[2] * B[4]  # indirect doctest
+            (1,3,2)*(2,3)
+        """
+        B = self._extalg.basis()
+        temp = B[x] * B[y]
+        return self.element_class(self, temp._monomial_coefficients)
+
+
+class Representation_Symmetric(Representation_abstract):
+    r"""
+    The symmetric power representation in a fixed degree.
+    """
+    def __init__(self, rep, degree, **options):
+        r"""
+        Initialize ``self``.
+
+        EXAMPLES::
+
+            sage: G = groups.matrix.GL(3, 2)
+            sage: R = G.regular_representation(side="right")
+            sage: S2 = R.symmetric_power(2)
+            sage: TestSuite(S2).run()
+            sage: S0 = R.symmetric_power(0)
+            sage: TestSuite(S2).run()
+
+            sage: R.symmetric_power(-2)
+            Traceback (most recent call last):
+            ...
+            ValueError: the degree must be a nonnegative integer
+            sage: R.symmetric_power(3/2)
+            Traceback (most recent call last):
+            ...
+            ValueError: the degree must be a nonnegative integer
+        """
+        from sage.rings.polynomial.polynomial_ring_constructor import PolynomialRing
+        from sage.combinat.integer_vector import IntegerVectors
+        from sage.rings.integer_ring import ZZ
+        self._degree = degree
+        self._rep = rep
+        R = rep.base_ring()
+        dim = rep.dimension()
+        if degree not in ZZ or degree < 0:
+            raise ValueError(f"the degree must be a nonnegative integer")
+        self._symalg = PolynomialRing(R, 'e', dim)
+        self._basis_order = list(rep.basis().keys())
+        G = self._symalg.gens()
+        self._inv_map = {b: G[i] for i, b in enumerate(self._basis_order)}
+        ind = IntegerVectors(degree, dim)
+        Representation_abstract.__init__(self, rep.semigroup(), rep.base_ring(), rep.side(),
```

Comment:
Could this be set up as a subobject of `Representation_Tensor`?
And the exterior power as a quotient?

Reply:
Yes, but it would make the code *significantly* more complicated and slower and we would only gain a certain amount of mathematical "purity." This really should stay as it is.

Reply:
OK